### PR TITLE
provider/aws: Convert VPN Gateway to upstream

### PR DESCRIPTION
**Pull request by catsby**

### CODE REVIEW

File: builtin/providers/aws/resource_aws_vpn_gateway.go [hunk 1]
```diff
@@ -270,8 +270,8 @@ func vpnGatewayStateRefreshFunc(conn *ec2.EC2, id string) resource.StateRefreshF
 			return nil, "", nil
 		}
 
-		vpnGateway := &resp.VPNGateways[0]
-		return vpnGateway, *vpnGateway.State, nil
+		vpnGateway := resp.VPNGateways[0]
+		return vpnGateway, "", nil
```
Comment (phinze) [inline builtin/providers/aws/resource_aws_vpn_gateway.go]: Aren't we supposed to return the state here?

**Comment by phinze** [comment]
LGTM generally, with 1 question inline.

File: builtin/providers/aws/resource_aws_vpn_gateway.go (hunk 1 above)
Comment (catsby) [inline builtin/providers/aws/resource_aws_vpn_gateway.go]: We were returning it, but never using it. It's kind of a funky remnant the original contributor got from copying LC. I should just overhaul the whole method...

**Comment by catsby** [comment]
@phinze I refactored to remove that refresh func that wasn't being used as a refresh func, take another look when you can

**Comment by phinze** [comment]
Cool LGTM :ship:

**Pull request merged**